internal/application/callback.go

```diff
@@ -53,6 +53,19 @@ func UpdateProfile(profileRequest requests.DeviceProfileRequest, dic *di.Contain
 	return nil
 }
 
+func DeleteProfile(profileName string, dic *di.Container) errors.EdgeX {
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	if cache.CheckProfileNotUsed(profileName) {
+		err := cache.Profiles().RemoveByName(profileName)
+		if err != nil {
+			errMsg := fmt.Sprintf("failed to remove device profile %s", profileName)
+			return errors.NewCommonEdgeX(errors.KindServerError, errMsg, err)
+		}
+	}
+	lc.Debugf("profile %s is removed from cache", profileName)
```

Comment:
if `cache.CheckProfileNotUsed` returns false, print a warning "received Profile Deletion System Event for %s(profileName), but the profile is still used by some devices"

Also, don't print `lc.Debugf("profile %s is removed from cache", profileName)` if `cache.CheckProfileNotUsed` returns false